src/orders/components/OrderTransaction/components/TransactionEvents/components/EventCreatedBy.test.tsx

```diff
@@ -0,0 +1,53 @@
+import { appAvatar } from "@saleor/apps/fixtures";
+import { appPath } from "@saleor/apps/urls";
+import { staffMemberAvatar } from "@saleor/staff/fixtures";
+import { staffMemberDetailsPath } from "@saleor/staff/urls";
+import Wrapper from "@test/wrapper";
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { MemoryRouter } from "react-router";
+
+import { EventCreatedBy } from "./EventCreatedBy";
+
+describe("EventCreatedBy", () => {
+  it("doesn't display anything if there's no app / user", () => {
+    render(<EventCreatedBy createdBy={null} />);
+    expect(screen.queryByRole("link")).not.toBeInTheDocument();
+  });
+
+  it("displays a link to the app if app is passed", () => {
+    render(
+      <MemoryRouter>
+        <Wrapper>
+          <EventCreatedBy createdBy={appAvatar} />
+        </Wrapper>
+      </MemoryRouter>,
```

Comment:
suggestion: try to test this component in solution and mock everything

Reply:
To be frank I don't see how it benefits our tests in any meaningful way - in the app we wrap the components in those providers from `Wrapper` anyway, just with real data + it sets the data to be always the one specified, so it doesn't differ between test runs. I think this is against the main philosophy of react-testing-library:

> "The more your tests resemble the way your software is used, the more confidence they can give you." [from KCD](https://twitter.com/kentcdodds/status/977018512689455106)

If we mock everything, then what's the point of writing tests?

We need `MemoryRouter` because otherwise we wouldn't be able to render `Link` component from react-router. Mocking `Link` instead of using this fake router fails to actually do any meaningful test - we want to test the real component if it uses correct `href`.

Reply:
In this case, you have to use Memory Router, sorry for confusion, but you can get rid of Wrapper as the test passed without it.